rails/etsy_product_catalog/app/models/category.rb

```diff
@@ -0,0 +1,12 @@
+class Category < ActiveRecord::Base
+	#has_many :listings
```

Comment:
I'd uncomment that line. A category *does* have many listings. You would need to account for the fact that the foreign key is etsy_category_id rather than just category_id, but it would still be better Rails code to handle it in a has_many association rather than the category_listings method, though that method is written correctly.